web/src/features/charts/FuturePrice.tsx

```diff
@@ -46,13 +47,37 @@ export function FuturePrice({ futurePrice }: { futurePrice: FuturePriceData | nu
     [filteredPriceData, granularity]
   );
 
-  if (!futurePrice) {
-    return null;
-  }
-
+  const priceBarContainerReference = useRef<HTMLDivElement>(null);
+  const [positiveWidth, setPositiveWidth] = useState(0);
+  const [negativeWidth, setNegativeWidth] = useState(0);
   const hasNegativePrice = minPriceTotal < 0;
   const negativePercentage = negativeToPostivePercentage(minPriceTotal, maxPriceTotal);
 
+  useEffect(() => {
+    const handleResize = () => {
+      const divWidth = priceBarContainerReference?.current?.getBoundingClientRect().width;
+      setPositiveWidth(calculateWidth(divWidth, negativePercentage, false));
+      setNegativeWidth(calculateWidth(divWidth, negativePercentage, true));
+    };
+
+    const observer = new ResizeObserver(handleResize);
+    const current = priceBarContainerReference.current;
+
+    if (current) {
+      observer.observe(current);
+    }
+
+    return () => {
+      if (current) {
+        observer.unobserve(current);
+      }
+    };
+  }, [priceBarContainerReference, negativePercentage, isCollapsed]);
```

Comment:
Do we really need a use effect?

It should only be used as a escape hatch if it can't fit in the normal rendering flow of react.

Wouldn't the useResizeObserver hook be a better option?

Reply:
I tried just using the useResizeObserver without the useEffect but I could not get it to work. If you have time and think it is important then it would be nice if you could try it. But you decide

Reply:
I can take a quick look but otherwise I don't consider this a blocking thing.